Parcel 2 increases output size by 12.5x when using `export * as x` on a `bundle-text`ed file

# 🐛 bug report

For some reason, parcel adds a bunch of nothing-doing boilerplate in some cases where it's not needed.

## 🎛 Configuration (.babelrc, package.json, cli command)

Is reproducable without config

## 🤔 Expected Behavior

Output:
```
(()=>{console.log("Hey","Ho")})();
//# sourceMappingURL=main.js.map
```

## 😯 Current Behavior

```
(()=>{var e="undefined"!=typeof globalThis?globalThis:"undefined"!=typeof self?self:"undefined"!=typeof window?window:"undefined"!=typeof global?global:{},r={},o={},n=e.parcelRequire94c2;null==n&&((n=function(e){if(e in r)return r[e].exports;if(e in o){var n=o[e];delete o[e];var t={id:e,exports:{}};return r[e]=t,n.call(t.exports,t,t.exports),t.exports}var i=new Error("Cannot find module '"+e+"'");throw i.code="MODULE_NOT_FOUND",i}).register=function(e,r){o[e]=r},e.parcelRequire94c2=n),n.register("3yjGo",(function(e,r){var o,n,t,i,l;o=e.exports,n="register",t=()=>l,i=e=>l=e,Object.defineProperty(o,n,{get:t,set:i,enumerable:!0,configurable:!0});var f={};l=function(e){for(var r=Object.keys(e),o=0;o<r.length;o++)f[r[o]]=e[r[o]]}})),n("3yjGo").register(JSON.parse('{"eMZYd":"main.js"}'));console.log("Hey","Ho")})();
//# sourceMappingURL=main.js.map
```

## 💁 Possible Solution

Don't output the wrapping stuff when not needed

## 🔦 Context

I was about to indulge myself in a precious compacted output bundle and was shocked to see such lazy bytes without a purpose.

## 💻 Code Sample

[parcel-bloat.zip](https://github.com/parcel-bundler/parcel/files/7068131/parcel-bloat.zip)

please run `yarn install` and then `yarn build`

Basically this is main.ts
```
import { a_html, b_html } from "./exporting_file";
console.log(a_html, b_html);
```

this is exporting_file.ts
```
export * as a_html from "bundle-text:./a.html";
export * as b_html from "bundle-text:./b.html";
```
and the html files contain `Hey` and `Ho` respectively.


## 🌍 Your Environment

<!--- Include as many relevant details about the environment you experienced the bug in -->

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           |2.0.0-nightly.811+1452a30c|
| Node             |v16.7.0|
| npm/Yarn         |3.0.1|
| Operating System |macOS 10.15.7 (19H1323)|

<!-- Love parcel? Please consider supporting our collective:
👉  https://opencollective.com/parcel/donate -->

--- a/packages/core/integration-tests/test/javascript.js
+++ b/packages/core/integration-tests/test/javascript.js
@@ -4045,6 +4045,53 @@ describe('javascript', function() {
       path.join(__dirname, '/integration/bundle-text/index.js'),
     );
 
+    assertBundles(b, [{type: 'js', assets: ['index.js']}]);
+
+    let jsBundle = b.getBundles().find(b => b.type === 'js');
+    console.log(await outputFS.readFile(jsBundle.filePath, 'utf8'));
+
+    let cssBundleContent = (await run(b)).default;
+
+    assert(
+      cssBundleContent.startsWith(
+        `body {
+  background-color: #000000;
+}
+
+.svg-img {
+  background-image: url("data:image/svg+xml,%3Csvg%3E%0A%0A%3C%2Fsvg%3E%0A");
+}`,
+      ),
+    );
+
+    assert(!cssBundleContent.includes('sourceMappingURL'));
+  });
+
+  it('should not include the runtime manifest for `bundle-text`', async () => {
+    let b = await bundle(
+      path.join(__dirname, '/integration/bundle-text/index.js'),
+      {
+        mode: 'production',
+        defaultTargetOptions: {shouldScopeHoist: false, shouldOptimize: false},
+      },
+    );
+
+    assertBundles(b, [
+      {
+        name: 'index.js',
+        type: 'js',
+        assets: ['esmodule-helpers.js', 'index.js'],
+      },
+      {
+        type: 'svg',
+        assets: ['img.svg'],
+      },
+      {
+        type: 'css',
+        assets: ['text.scss'],
+      },
+    ]);
+
     let cssBundleContent = (await run(b)).default;
 
     assert(

--- a/packages/core/integration-tests/test/javascript.js
+++ b/packages/core/integration-tests/test/javascript.js
@@ -4045,11 +4045,6 @@ describe('javascript', function() {
       path.join(__dirname, '/integration/bundle-text/index.js'),
     );
 
-    assertBundles(b, [{type: 'js', assets: ['index.js']}]);
-
-    let jsBundle = b.getBundles().find(b => b.type === 'js');
-    console.log(await outputFS.readFile(jsBundle.filePath, 'utf8'));
-
     let cssBundleContent = (await run(b)).default;
 
     assert(

--- a/packages/runtimes/js/src/JSRuntime.js
+++ b/packages/runtimes/js/src/JSRuntime.js
@@ -226,7 +226,9 @@ export default (new Runtime({
 
     if (
       shouldUseRuntimeManifest(bundle, options) &&
-      bundleGraph.getChildBundles(bundle).length > 0 &&
+      bundleGraph
+        .getChildBundles(bundle)
+        .some(b => b.bundleBehavior !== 'inline') &&
       isNewContext(bundle, bundleGraph)
     ) {
       assets.push({

